### PR TITLE
Autocomplete with batman syntax support

### PR DESCRIPTION
**Pull request by suchitadoshi1987**
This PR depends on #204 

TODO:

- [x]  Add an option in the config for batman support
- [x]  Update script completion providers
- [x]  Add tests

### CODE REVIEW

**Comment by suchitadoshi1987** [comment]
@lifeart I want to test the server based on some config values that the users set in their VSCode's `settings.json`. What is the best way to test that?

File: src/utils/layout-helpers.ts [hunk 1]
```diff
@@ -364,9 +375,9 @@ export function getProjectAddonsInfo(root: string) {
     }
 
     if (version === 1) {
-      // log('isEmberAddon', packagePath);
+      const addonName = packagePath.split('/').pop();
```
Comment (lifeart) [inline src/utils/layout-helpers.ts:380]: addon name could be namespaced like `@my-company/my-addon`, we have to handle it somehow or skip this case for now (check for `@` at first char)

File: src/utils/common-helpers.ts [hunk 2]
```diff
@@ -7,3 +8,14 @@ export async function getAppRootFromConfig(server: Server) {
     return Promise.resolve('');
   }
 }
+
+export const mProjectRoot = memoize(getProjectParentRoot);
+
+/**
+ * Find the top level root of the project.
+ */
+export function getProjectParentRoot(root: string, appRoot: string) {
```
Comment (lifeart) [inline src/utils/common-helpers.ts]: cold we use projectForUri(uri).root for this needs? or it's some different cases?

Comment (lifeart) [reply src/utils/common-helpers.ts]: https://github.com/lifeart/ember-language-server/blob/component-context-info-origin/src/project-roots.ts#L100 root definition

File: src/utils/layout-helpers.ts (hunk 1 above)
Comment (suchitadoshi1987) [reply src/utils/layout-helpers.ts:380]: good point

File: src/utils/layout-helpers.ts [hunk 3]
```diff
@@ -482,7 +495,13 @@ export function builtinModifiers(): CompletionItem[] {
   ];
 }
 
-export function listComponents(_root: string): CompletionItem[] {
+function hasNamespaceSupport(root: string) {
```
Comment (lifeart) [inline src/utils/layout-helpers.ts:498]: could we add caching to this function, for 10 minutes or such, don't think this value may change very often

**Pull request merged**